packages/skia/cpp/api/JsiSkSurface.h

```diff
@@ -28,11 +29,29 @@ namespace RNSkia {
 namespace jsi = facebook::jsi;
 
 class JsiSkSurface : public JsiSkWrappingSkPtrHostObject<SkSurface> {
+private:
+  ThreadSafeDeletion<SkSurface> _deletionHandler;
+
 public:
   JsiSkSurface(std::shared_ptr<RNSkPlatformContext> context,
                sk_sp<SkSurface> surface)
       : JsiSkWrappingSkPtrHostObject<SkSurface>(std::move(context),
-                                                std::move(surface)) {}
+                                                std::move(surface)),
+        _deletionHandler() {
+    // Drain any pending deletions when creating new surfaces
+    ThreadSafeDeletion<SkSurface>::drainDeletionQueue();
+  }
+
+  ~JsiSkSurface() override {
+    // Handle thread-safe deletion
+    auto objectToDelete = _deletionHandler.handleDeletion(getObject());
+    if (!objectToDelete) {
+      // Object was queued for deletion on another thread
+      // Clear it to prevent base class destructor from deleting it
+      setObject(nullptr);
+    }
```

Comment:
imo it'd be cleaner if `handleDeletion` would be the only place that controls the object destruction as opposed to having an implicit contract via return value. This would be less error prone as now you need to understand the implementation and meaning of the return value in order to handle it correctly. So instead of doing that you'd always call `setObject(nullptr)` and then `handleDeletion` would conditionally delete it.